specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,454 @@
+
+    Note over Alice,SBT_Registry: After a while Alice decides to merge her `alice2` account into `alice2`
+    Alice->>SBT_Registry: sbt_soul_transfer(alice) --accountId alice2
+
+    Alice-->>+SBT_Registry: sbt_tokens_by_owner(alice2)
+    SBT_Registry-->>-Alice: []
+
+    Alice-->>+SBT_Registry: sbt_tokens_by_owner(alice)
+    SBT_Registry-->>-Alice: [[SBT_1_Contract, [238]], [SBT_2_Contract, [7991, 7992]]]}
+```
+
+## Consequences
+
+### Positive
+
+- Template and set of guidelines for creating SBT tokens.
+- Ability to create SBT aggregators.
+- Ability to use SBT as a primitive to model non KYC identity, badges, certificates etc...
+- SBT can be further used for "lego" protocols, like: Proof of Humanity (discussed for NDC Governance), undercollateralized lending, role based authentication systems, innovative economic and social applications...
+- Standard recoverability mechanism.
+- SBT are considered as a basic primitive for Decentralized Societies.
+- new way to implement Sybil attack resistance.
+
+### Neutral
+
+- The API partially follows the NEP-171 (NFT) standard. The proposed design is to have native SBT API and also support NFT based queries.
+  NOTE: we can decide to use `nft_` prefix whenever possible.
+
+### Negative
+
+- new set of events to be handled by the indexer and wallets.
+- complexity of integration with a registry: all SBT related transactions must go through Registry.
+
+## Considerations
+
+Being fully compatible with NFT standard is a desirable. However, given the requirements related to _soul transfer_ we didn't find an applaudable solution. Also we decided to use u64 as a Token ID, diverging further from the NFT NEP-171 standard.
+
+Give that our requirements are much striker, we need to reconsider the level of compatibility with NEP-171 NFT.
+There are so many examples where NFT standards are poorly or improperly implemented, adding another standard with differing functionality but equal naming in there will cause lots of misclassifications between NFTs/SBTs, and then recover methods called on NFT contracts, SBTs attempted to be listed on NFT marketplaces and so on.
+
+CALL FOR ACTION: Shall the events be issued by the registry or by the issuing contract?
+
+- registry: we query registry, so it makes sense to use registry. If we emit the event by the registry, we need to add smart contract argument as a field of the events.
+- maybe we should reduce amount of tokens: merge {mint, renew}?
+- Should we use NEP-171 Mint and NEP-171 Burn (instead of revoke) events? If the events will be emitted by registry, then we need new events to include the contract address. Since the native token ID is different in SBT and NFT, we should probably keep SBT events version. Also, with registry, it makes less sense to issue NEP-171 Mint and Burn.
+- Confirm that only the registry emits the events.
+- Decide if we need to keep memo in the events (it's already in the transaction).
```

Comment:
From the indexing point of view, it is easier to use a single source of truth as events could be generated from batch calls or even some internal implementation of some other method, and then trying to parse the function call arguments is going to be an unnecessary burden, so it is better to include `memo` in the events.

Reply:
My general advice for all events definition is to keep it to minimum:
* must uniquely identify a transaction given a state transition of an object we are monitoring (memo doesn't define a state transition).
* must reflect the use-case (correct event type).
Memo is a universal argument, and doesn't define a use-case nor the object.